packages/devtools_app/lib/src/code_size/code_size_screen.dart

```diff
@@ -157,149 +108,79 @@ class CodeSizeBodyState extends State<CodeSizeBody>
-    return [
-      snapshotRoot == null
-          ? _buildSnapshotTabBody()
-          : _buildTreemapAndTableSplitView(showDiff: false),
-      diffRoot == null
-          ? _buildDiffTabBody()
-          : _buildTreemapAndTableSplitView(showDiff: true),
-    ];
-  }
+  CodeSizeController controller;
 
-  Widget _buildSnapshotTabBody() {
-    return Padding(
-      padding: const EdgeInsets.all(16.0),
-      child: Column(
-        crossAxisAlignment: CrossAxisAlignment.stretch,
-        children: [
-          Expanded(
-            child: FileImportContainer(
-              title: 'Snapshot',
-              onImport: () {
-                controller.loadSnapshotFile(
-                  '$current/lib/src/code_size/stub_data/new_v8.json',
-                );
-              },
-              importedFile: controller.snapshotFile.value,
-            ),
-          ),
-          const SizedBox(height: 16.0),
-          _builHelpText(),
-        ],
-      ),
-    );
+  TreemapNode snapshotRoot;
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+
+    final newController = Provider.of<CodeSizeController>(context);
+    if (newController == controller) return;
+    controller = newController;
+
+    snapshotRoot = controller.snapshotRoot.value;
+    addAutoDisposeListener(controller.snapshotRoot, () {
+      setState(() {
+        snapshotRoot = controller.snapshotRoot.value;
+      });
+    });
   }
 
-  Widget _buildDiffTabBody() {
-    return Padding(
-      padding: const EdgeInsets.all(16.0),
-      child: Column(
-        crossAxisAlignment: CrossAxisAlignment.stretch,
-        children: [
-          Expanded(
-            child: Row(
-              children: [
-                Expanded(
-                  child: FileImportContainer(
-                    title: 'Old Snapshot',
-                    onImport: () {
-                      controller.loadOldDiffSnapshotFile(
-                        '$current/lib/src/code_size/stub_data/old_v8.json',
-                      );
-                    },
-                    importedFile: controller.diffOldSnapshotFile.value,
-                  ),
-                ),
-                const SizedBox(width: 16.0),
-                Expanded(
-                  child: FileImportContainer(
-                    title: 'New Snapshot',
-                    onImport: () {
-                      controller.loadNewDiffSnapshotFile(
-                        '$current/lib/src/code_size/stub_data/new_v8.json',
-                      );
-                    },
-                    importedFile: controller.diffNewSnapshotFile.value,
-                  ),
-                ),
-              ],
-            ),
-          ),
-          const SizedBox(height: 16.0),
-          _builHelpText(),
-        ],
-      ),
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      children: [
+        buildSnapshotViewControls(),
+        Expanded(child: buildBody()),
```

Comment:
instead of extracting to buildBody, just do this directly here:

`child: snapshotRoot != null ? _buildTreemapAndTableSplitView() : _buildImportSnapshotView(),`